moped-editor/src/views/projects/projectView/ProjectFundingTable.js

```diff
@@ -305,25 +294,233 @@ const ProjectFundingTable = () => {
     setSnackbarState(DEFAULT_SNACKBAR_STATE);
   };
 
-  /**
-   * Column configuration for <MaterialTable>
-   */
-  const columns = [
+  const handleRowModesModelChange = (newRowModesModel) => {
+    setRowModesModel(newRowModesModel);
+  };
+
+  // adds a blank row to the table and updates the row modes model
+  const handleAddRecordClick = () => {
+    // use a random id to keep track of row in row modes model and data grid rows
+    // before the record is added to the db
+    const id = Math.floor(Math.random() * 10000);
+    setRows((oldRows) => [
+      ...oldRows,
+      {
+        id,
+        funding_source_id: null,
+        funding_program_id: null,
+        funding_description: null,
+        funding_status_id: null,
+        fund: null,
+        dept_unit: null,
+        funding_amount: null,
+        isNew: true,
+        proj_funding_id: id,
+      },
+    ]);
+    setRowModesModel((oldModel) => ({
+      ...oldModel,
+      [id]: { mode: GridRowModes.Edit, fieldToFocus: "source" },
+    }));
+  };
+
+  const handleEditClick = (id) => () => {
+    setRowModesModel({ ...rowModesModel, [id]: { mode: GridRowModes.Edit } });
+  };
+
+  const handleSaveClick = (id) => () => {
+    setRowModesModel({ ...rowModesModel, [id]: { mode: GridRowModes.View } });
+  };
+
+  // handles row delete
+  const handleDeleteClick = (id) => () => {
```

Comment:
There is a confirmation before delete on this table in production. Do y'all think we should keep that? Amenity commented on the confirmation that pops up when deleting a tag from the summary view, and pointed out that it isn't a critical action and that it can be viewed in the activity log if it was a mistake. In this case, it would be harder to restore the deleted record since the activity log doesn't show the record details.

I found myself looking at this article which has some good guidelines too. https://www.nngroup.com/articles/confirmation-dialog/

Reply:
received good support for using `DeleteConfirmationModal` here so i added to https://github.com/cityofaustin/atd-data-tech/issues/16831